cookbooks/aws-parallelcluster-config/recipes/finalize.rb

```diff
@@ -25,3 +25,7 @@
 
 include_recipe 'aws-parallelcluster-scheduler-plugin::finalize' if node['cluster']['scheduler'] == 'plugin'
 include_recipe 'aws-parallelcluster-slurm::finalize' if node['cluster']['scheduler'] == 'slurm'
+
+# Log the uptime at the end of running chef scripts.
+uptime_ms = IO.read('/proc/uptime').split[0].to_f * 1000
+printf("Recipes Finish Uptime: %f\n", uptime_ms)
```

Comment:
I'm wondering if this printf is evaluated at the end of the chef compile (as we probably would like it to be) or during converge phase (meaning the chef run is not over yet).

Also since the main bootstrap orchestration is managed by cloud-init an idea could be to use the timestamps present in `/var/log/cloud-init.log`

Reply:
When I ran it in tests it was one of the last print outs -- so it seemed to capture most of the work done in chef being completed.

I'll have to check -- but we could also leverage those depending on what we're trying to capture, however the point of this print is to have the running time, not a timestamp as it's easier on the cloudwatch side of things to not have to compute the difference between timestamps (which are absolute). Unless I'm misunderstanding what is in the cloud-inIt log

Reply:
I believe that print is executed in the compile phase. if so we are printing the uptime before the finalize phase runs. Consider that in case the cluster has static capacity the finalize converge phase could take minutes.

We probably can either use the log resource (https://docs.chef.io/resources/log/) or the execute one (https://docs.chef.io/resources/execute/)

Reply:
Also notice that at the end of the ` cloud-init-output.log`
there is already info about execution time, both absolute timestamp and elapsed time, e.g.
```
Cloud-init v. 19.3-44.amzn2 finished at Mon, 20 Dec 2021 14:14:23 +0000. Datasource DataSourceEc2.  Up 166.94 seconds
```

Reply:
Ah -- I will close this PR in favor of leveraging that data point from `cloud-init`.